internal/internal_workflow_execution_options.go

```diff
@@ -144,6 +154,17 @@ func versioningOverrideFromProto(versioningOverride *workflowpb.VersioningOverri
 	}
 }
 
+func onConflictOptionsToProto(onConflictOptions *OnConflictOptions) *workflowpb.OnConflictOptions {
```

Comment:
nit: this could be a method on the `OnConflictOptions` struct but I'd just use the proto directly here since we're not exposing this API to users yet.